src/mlpack/tests/main_tests/linear_svm_test.cpp

```diff
@@ -0,0 +1,853 @@
+  const arma::mat parameters1 = std::move(
+      CLI::GetParam<LinearSVMModel*>("output_model")->svm.Parameters());
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("optimizer", std::string("psgd"));
+  SetInputParam("num_classes", int(2));
+  SetInputParam("step_size", double(1.02));
+
+  // Second solution.
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after second training.
+  const arma::mat& parameters2 =
+      CLI::GetParam<LinearSVMModel*>("output_model")->svm.Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+ * Ensuring that tolerance has some effects on the output.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMDiffToleranceTest)
+{
+  arma::mat trainData = "2 0 0;"
+                        "0 0 0;"
+                        "0 2 1;"
+                        "1 0 2;"
+                        "0 1 0";
+
+  arma::Row<size_t> trainLabels = "1 0 1";
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+  SetInputParam("optimizer", std::string("psgd"));
+  SetInputParam("num_classes", int(2));
+  SetInputParam("tolerance", double(1e-1));
+
+  // First solution.
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after first training.
+  const arma::mat parameters1 = std::move(
+      CLI::GetParam<LinearSVMModel*>("output_model")->svm.Parameters());
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("optimizer", std::string("psgd"));
+  SetInputParam("num_classes", int(2));
+  SetInputParam("tolerance", double(1e-10));
+
+  // Second solution.
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after second training.
+  const arma::mat& parameters2 =
+      CLI::GetParam<LinearSVMModel*>("output_model")->svm.Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+ * Ensuring that lbfgs optimizer converges to a different result than psgd.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMDiffOptimizerTest)
+{
+  arma::mat trainData = "2 0 0;"
+                        "0 0 0;"
+                        "0 2 1;"
+                        "1 0 2;"
+                        "0 1 0";
+
+  arma::Row<size_t> trainLabels = "1 0 1";
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+  SetInputParam("optimizer", std::string("lbfgs"));
+
+  // First solution.
+  mlpackMain();
```

Comment:
Here, too, we will need a fixed random seed to ensure that both optimizers start from the same initial point.

Reply:
Done.